fabtests/component/sock_test.c

```diff
@@ -0,0 +1,375 @@
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER const AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS const THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <sys/time.h>
+#include <netdb.h>
+#include <time.h>
+#include <poll.h>
+#include <sys/epoll.h>
+#include <errno.h>
+
+#include <shared.h>
+
+int *clients;
+
+fd_set select_set;
+int max_sock, connections;
+struct pollfd *poll_set;
+struct epoll_event *events;
+int efpd;
+
+static int start_server()
+{
+	struct addrinfo *ai = NULL;
+	int ret;
+	int optval = 1;
+
+	ret = getaddrinfo(opts.src_addr, opts.src_port, NULL, &ai);
+	if (ret) {
+		printf("getaddrinfo failed\n");
+		goto out;
+	}
+
+	listen_sock = socket(ai->ai_family, SOCK_STREAM, 0);
+	if (listen_sock < 0) {
+		printf("socket error: %i: %s\n", errno, strerror(errno));
+		ret = -1;
+		goto out;
+	}
+
+	ret = setsockopt(listen_sock, SOL_SOCKET, SO_REUSEADDR, (char *) &optval, 
+		sizeof(optval));
+	if (ret) {
+		printf("error setting socket options\n");
+		goto out;
+	}
+
+	ret = bind(listen_sock, ai->ai_addr, ai->ai_addrlen);
+	if (ret) {
+		printf("bind failed\n");
+		goto out;
+	}
+
+	ret = listen(listen_sock, connections);
+	if (ret) {
+		printf ("listen failed\n");
+		goto out;
+	}
+
+	freeaddrinfo(ai);
+	return ret;
+
+out:
+	close(listen_sock);
+	freeaddrinfo(ai);
+	return ret;
+}
+
+static int server_connect()
+{
+	int new_sock, i;
+
+	for (i = 0; i < connections; i++) {
+		new_sock = accept(listen_sock, NULL, NULL);
+		if (new_sock < 0) {
+			printf("error during server init\n");
+			return -1;
+		}
+
+		clients[i] = new_sock;
+		
+	}
+	return 0;
+}
+
+static int client_connect()
+{
+	struct addrinfo *res;
+	int i, ret, new_sock;
+
+	ret = getaddrinfo(opts.dst_addr, opts.dst_port, NULL, &res);
+	if (ret) {
+		printf("getaddrinfo failed\n");
+		goto out;
+	}
+
+	for (i = 0; i < connections; i++) {
+
+		new_sock = socket(res->ai_family, SOCK_STREAM, 0);
+		if (new_sock < 0) {
+			ret = -1;
+			goto out;
+		}
+
+		ret = connect(new_sock, res->ai_addr, res->ai_addrlen);
+		if (ret)
+			goto out;
+
+		clients[i] = new_sock;
+	}
+
+out:
+	freeaddrinfo(res);
+	return ret;
+}
+
+static int init_select() 
+{
+	FD_ZERO(&select_set);
+	FD_SET(listen_sock, &select_set);
+	max_sock = listen_sock;
+
+	return 0;
+}
+
+static int time_select() 
+{
+	int i, j;
+
+	ft_start();
+	for (i = 0; i < opts.iterations; i++) {
+		for (j = 0; j < connections; j++) {
+			FD_SET(clients[j], &select_set);
+
+			if (clients[j] > max_sock)
+				max_sock = clients[j];
+		}
+	}
+	ft_stop();
+	
+	printf("%i: %f\n", connections, (double)(end.tv_nsec + end.tv_sec * 
+			1000000000) - (start.tv_nsec + start.tv_sec * 1000000000)
+			/ connections);
```

Comment:
Create a function to display the results, so we can avoid duplicating the calculations in each place.